Create new branch and upload the newest version of hyp1.py

CoitThomas opened the pull request:
God I hope this works...

CoitThomas committed:
Create new branch and upload the newest version of hyp1.py

CoitThomas committed:
Add unit tests.

CoitThomas committed:
Pull the body of hyp1.py into its own module, 'find_hypotenuse.py', for unit testing purposes.

CoitThomas committed:
Remove print function from hyp1.py.
Simplify and move print function to find_hypotenuse as a single line.
Add final line to unit tests.

austinwiltshire commented on design_0/hyp1.py:
```diff
@@ -1,26 +1,38 @@
+    triangle.
+    """
+    print 'Enter the two sides of your right triangle.'
+    input_a = input('a:')
```
https://docs.python.org/2/library/functions.html#input

Use raw_input. Python3 fixes the safety issues here, but I get pylint errors on this line.

austinwiltshire replied:
raw input  will also bring in a string. 

http://stackoverflow.com/questions/379906/parse-string-to-float-or-int

just pull in the string and cast to a float there.

austinwiltshire replied:
given you'll be converting at the outset to a float, you may not even need your assert and convert, you'd only need the assert > 0 after that point.

austinwiltshire commented on design_0/test_assert_and_convert.py:
```diff
@@ -0,0 +1,5 @@
+from hyp1 import assert_and_convert
```
needs a module docstring

austinwiltshire commented on design_0/test_assert_and_convert.py:
```diff
@@ -0,0 +1,5 @@
+from hyp1 import assert_and_convert
+
+def test_assert_and_convert():
```
needs a function docstring. usually these provide information on what you're testing

austinwiltshire commented on design_0/test_find_hypotenuse.py:
```diff
@@ -0,0 +1,5 @@
+from hyp1 import find_hypotenuse
```
module docstring

austinwiltshire commented on design_0/test_find_hypotenuse.py:
```diff
@@ -0,0 +1,5 @@
+from hyp1 import find_hypotenuse
+
+def test_find_hypotenuse():
```
test docstring

austinwiltshire commented on design_0/find_hypotenuse.py:
```diff
@@ -0,0 +1,13 @@
+
+import hyp1
+
+(SIDE_A, SIDE_B) = hyp1.get_input()
```
http://stackoverflow.com/questions/419163/what-does-if-name-main-do 

Refactor this to be importable

austinwiltshire commented on design_0/hyp1.py:
```diff
@@ -1,26 +1,38 @@
-import math, os
+"""This module contains all the functions needed to successfully run
```
this filename is bad, find a more readable one

austinwiltshire replied:
(hint: use git mv for renaming files)

austinwiltshire commented on design_0/hyp1.py:
```diff
@@ -1,26 +1,38 @@
+    that input into a float. Return the input.
+    """
+    assert (isinstance(user_input, float) or
+            isinstance(user_input, int)), 'Value must be of type int or type float.'
```
drop support for int, YAGNI.

austinwiltshire commented on design_0/hyp1.py:
```diff
@@ -1,26 +1,38 @@
+    side_a = assert_and_convert(input_a)
+    input_b = input('b:')
+    side_b = assert_and_convert(input_b)
+    return (side_a, side_b)
```
nice use of tuple return

austinwiltshire commented on design_0/hyp1.py:
```diff
@@ -1,26 +1,38 @@
+
+    Calculate and return the hypotenuse of a right triangle.
+    """
+    return math.sqrt(math.pow(side_a, 2) + math.pow(side_b, 2))
```
doesn't side_a and side_b need to be positive for this to work?

austinwiltshire commented on design_0/test_find_hypotenuse.py:
```diff
@@ -0,0 +1,5 @@
+
+def test_find_hypotenuse():
+    assert find_hypotenuse(3, 4) == 5
+    assert 5.7 < find_hypotenuse(3.5, 4.5) < 5.71
```
nice test!

austinwiltshire replied:
what would happen if i ran find_hypotenuse(0, 0), or find_hypotenuse(-1, -5), or find_hypotenuse("hello", "world") or find_hypotenuse(None, True)?

austinwiltshire commented on design_0/test_get_input.py:
```diff
@@ -0,0 +1,4 @@
+from hyp1 import assert_and_convert, get_input
+
+def test_get_input():
```
we'll do a live session to talk about how to possibly test this.

CoitThomas committed:
Refactor and combine hyp1.py/hypotenuse_resources.py and find_hypotenuse.py.
Change input() to raw_input().
Refactor assert_and_convert.py. Change to convert_and_assert.py.
Add module docstrings and function docstrings.

CoitThomas committed:
Split convert_and_assert() into two seperate helper functions.
Update references to convert_and_assert().
Update test_get_input.py to adhere to pylint guidelines.
